File: Ashish/Week 2/3sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        a = []
        for i in range(len(nums)-1):
            j = i + 1
            k = len(nums) - 1
            while j < k:
                if nums[i] + nums[j] + nums[k] == 0:
                    a.append((nums[i], nums[j], nums[k]))
                    j += 1
                    k -= 1
                elif nums[i] + nums[j] + nums[k] > 0:
                    k -= 1
                else:
                    j += 1
        return set(a)

File: Farhaan/Week-2 Problems/3sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> output ;
        sort(nums.begin(),nums.end());
        for(int i=0; i<nums.size();i++){
            int left = i+1 ;
            int right = nums.size()-1;
            while(left < right){
                if(nums[i]+nums[left]+nums[right] == 0){
                    output.push_back({nums[i],nums[left],nums[right]});
                    int x = nums[left] ;
                    int y = nums[right] ;
                    while(left<right && nums[left] ==x){
                        left++ ;
                    }
                    while(left<right && nums[right] ==y){
                        right-- ;
                    }
                }else if(nums[i]+nums[left]+nums[right] > 0){
                    right --;
                }else {
                    left ++;
                }
            }
            while(i+1 < nums.size() && nums[i] == nums[i+1]){
                i++;
            }
        }
        return output ;
    }
};